### ISSUE TITLE
AWS amplify is just uber confusing. I was up and running with Firebase in 30 minutes.

### ISSUE THREAD

**Issue by dolphin54** [aws-amplify/docs#263]
I have to admit, AWS Amplify is confusing, just as Cognito is, the effort and time to get up and running is huge (putting off many people). I'm still trying to work out how to do things. All I wanted was to use the authorisation and authentication of AWS Amplify. I have ran the various commands amplify configure and init (which took a while to find on the site). All I know is some stack has been created and there is a user with full admin rights that has also been created!  I look at cognito, i see no user pools or anything.  Think I have spent the whole day on this now, getting very tempted to switch back to Firebase, sorry to say it was minutes with firebase.

**Comment by JSafaiyeh** [comment]
I came across this too. I understand CloudFormation handles all the resources; however, it does not give me the easy customization of resources as I would like.

I took the manual route and set up the resources myself and use Amplify for the SDK rather than using it to build my resources.

**Comment by muellerfr** [comment]
Hello @dolphin54,

sorry to hear that you are finding Amplify difficult to use. Which documentation page are you trying to follow? Also, which platform(s) are you building for? (iOS, Android, React Native or Web).

**Comment by dolphin54** [comment]
I am following this:
https://aws-amplify.github.io/docs/js/tutorials/building-ionic-4-apps/
but also the pages around it all.
(this page should really have some links to the pages about setting up the AWS side first).

I will be using it for my web and ionic apps.

There are basically two sides to this, the setup on AWS which should be agnostic as to what front end you use and then the specific bits from your front-end framework.

I have managed to get the ampilfy to do some auth setup, which in all essence should be one of the first things that amplify should ask you before even setting up an S3 bucket for downloads.
Took a while to find it:
amplify add auth
Now this does not give you much control, it has defaulted to a username and password, you go to cognito and you can't even change it to email, password or even username/email and password.

I know the amplify has setup two roles, anonymous user and authenticated user, How does one even know in other parts of your code, which one a user is.  Also how does one have an admin user for example and that to be identifiable in my Java backend?  Is there a nice way for the java to use the token and ask cognito for the user type? or is it all in the token?  Is there a Java plugin for this?

This is serious hard work to find and do anything.  I'm pretty sure where amplify is trying to aim for but it can made 20 times easier than this.

Also I would not like the user created by amplify to have full admin rights, can you let us have a policy json for the restricted ones it needs.

Thanks

**Comment by flyingace** [comment]
@muellerfr I'm going to jump in here to second the notion that the documentation is unclear at many points. I have been using the Authentication component on a react app my team is using and while doing a POC without styling or using custom components was very easy, once we started trying to do anything slightly more complex or even configure our Authenticator we have found ourselves stymied at several turns. Combine the spotty documentation with a lack of active user forums and you end up with a very frustrating developer experience.

I have opened a separate issue regarding an issue I am having trying to understand how to do something in Authenticator here: #269
